Source/WebCore/platform/graphics/texmap/coordinated/GraphicsLayerCoordinated.cpp

```diff
@@ -1092,8 +1057,11 @@ void GraphicsLayerCoordinated::commitLayerChanges(CommitState& commitState, floa
     if (m_pendingChanges.contains(Change::ContentsColor))
         m_platformLayer->setContentsColor(m_contentsColor);
 
-    if (m_pendingChanges.contains(Change::DirtyRegion))
-        updateDirtyRegion(commitState.visibleRect.size());
+    if (m_pendingChanges.contains(Change::DirtyRegion)) {
+        ASSERT(m_dirtyRegion.has_value());
+        m_platformLayer->setDirtyRegion(WTFMove(*m_dirtyRegion));
+        m_dirtyRegion = std::nullopt;
```

Comment:
```suggestion
        m_platformLayer->setDirtyRegion(*std::exchange(m_dirtyRegion, std::nullopt));
```

I've checked this idiom, and it looks better than my first suggestion. Using std::exchange produces a prvalue T (the optional), then dereferencing it gets us an xvalue. 

I've done some grepping and this idiom is used already in the code: see with `git grep "*std::exchange" | grep std::nullopt`

Reply:
Ok, updated it in both places. Thanks!